packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -2880,7 +2880,7 @@ function commitMutationEffectsOnFiber(
         }
 
         if (isHidden) {
-          if (!wasHidden) {
+          if (current !== null && current.memoizedState === null) {
```

Comment:
so my understanding is that the issue is happening bc when the hidden offscreen component is mounted, we are not running layout effects.

conceptually, why would we want to run layout effects if the component is hidden? what is the example of breaking behavior?

Reply:
> so my understanding is that the issue is happening bc when the hidden offscreen component is mounted, we are not running layout effects.

The issue is that layout effects are cleaned up even when Offscreen is hidden. If you look at the test, it may happen when two OffScreens are nested (keep in mind every Suspense boundary uses Offscreen).